tools/distro-json-generate.mjs

```diff
@@ -1,40 +1,69 @@
+import https from 'https';
 import fs from 'fs-extra';
 import shell from 'shelljs';
 
-shell.echo(`Verifying required packages...`);
+let url = 'https://debian.pages.debian.net/distro-info-data/ubuntu.csv';
 
-if (!shell.which(`distro-info`)) {
-  shell.echo('This script requires distro-info, exiting...');
-  shell.exit(2);
+/**
+ * @param {string} url
+ * @param {function} cb
+ */
+function csv(url, cb) {
+  https.get(url, (response) => {
+    let buffer = '';
+    response
+      .on('data', (d) => {
+        buffer += d.toString();
+      })
+      .on('end', () => {
+        cb(buffer);
+      })
+      .on('error', (e) => {
+        shell.echo(
+          `csv - Error fetching distro data from URL:${url} e: ${e}\n Exiting...`
+        );
+        shell.exit(2);
+      })
+      .setEncoding('utf8');
+  });
 }
 
-if (!shell.which(`sed`)) {
-  shell.echo('This script requires sed, exiting...');
-  shell.exit(2);
-}
+/**
+ * @param {string} raw
+ * @returns {string}
+ */
+function csvToJson(raw) {
+  const lines = raw.split(/\r?\n/);
+  const res = {};
+  const headers = lines[0].split(',');
 
-shell.echo(`OK`);
+  // drop headers
+  lines.shift();
 
-const ubuntuDistroInfo = shell.exec(
-  `ubuntu-distro-info --all -f | sed -r 's/Ubuntu|"|LTS |Debian //g; s/([0-9]+.[0-9]+) /\\1 /; s/.*/\\L&/; s/( [a-z]*) [a-z]*/\\1/g; s/^[ \\t]*//'`,
-  { silent: true }
-);
+  // drop "version" header
+  headers.shift();
 
-/**
- * @param {string} str
- * @returns {{}}
- */
-function objectify(str) {
-  let obj = {};
+  lines
+    // drop empty lines
+    .filter((l) => !!l)
+    .map((l) => {
+      const obj = {};
+      const line = l.split(',');
+      // eslint-disable-next-line
+      // @ts-ignore
```

Comment:
no, use jsdoc comments to declare types.

Reply:
switched to ts.

Reply:
Still an an issue

Reply:
ah, finally got it :|. 
had a miss consumption about `jsdoc`, thought of if as sort of  `javadoc` used to comment function prototypes only. 

fixed.